dimod/constrained.py

```diff
@@ -861,6 +987,14 @@ def vartype(self, v: Variable) -> Vartype:
         """The vartype of the given variable."""
         return self.objective.vartype(v)
 
+    def violations(self, sample_like, *, skip_satisfied: bool = False, clip: bool = False,
```

Comment:
I wonder if we should also accept a `tolerance` parameter for `clip`. But a better place for it is in a method such as `is_feasbile`. Can we add a `is_feasible` method in this PR or should I make a new issue for it?